src/governance_token.cairo

```diff
@@ -50,15 +50,35 @@ mod GovernanceToken {
     use super::{IERC20, IGovernanceToken, ContractAddress};
     use traits::{Into, TryInto};
     use option::{OptionTrait};
-    use starknet::{get_caller_address, get_block_timestamp};
+    use starknet::{get_caller_address, get_block_timestamp, StorePacking};
     use zeroable::{Zeroable};
+    use integer::{u256_safe_divmod, u256_as_non_zero};
 
-    #[derive(Copy, Drop, starknet::Store)]
+    #[derive(Copy, Drop, PartialEq)]
     struct DelegatedSnapshot {
         timestamp: u64,
         delegated_cumulative: u256,
     }
 
+    const TWO_POW_64: u128 = 0x10000000000000000_u128;
+
+    impl DelegatedSnapshotStorePacking of StorePacking<DelegatedSnapshot, felt252> {
+        fn pack(value: DelegatedSnapshot) -> felt252 {
+            (value.delegated_cumulative + u256 {
```

Comment:
if `delegated_cumulative` is actually always less than 256bits shouldn't you change its type? Or is `u128`not large enough?

Reply:
it can be up to 192 bits, there is no u192, but in general i use bigger types in a lot of places and enforce restrictions at the storage boundary with cairo, because it doesn't make sense to enforce these value restrictions anywhere else